server/services/v1/search_indexer.go

```diff
@@ -179,12 +181,29 @@ func PackSearchFields(data *internal.TableData, collection *schema.DefaultCollec
 
 	decData = FlattenObjects(decData)
 
-	// now if there is any array we need to pack it
-	for key, value := range decData {
-		if _, ok := value.([]any); ok {
-			// pack any array field
-			if decData[key], err = jsoniter.MarshalToString(value); err != nil {
-				return nil, err
+	// pack any date time or array fields here
+	for _, f := range collection.QueryableFields {
+		key, value := f.Name(), decData[f.Name()]
+		if value == nil {
+			continue
+		}
+		if f.ShouldPack() {
+			switch f.DataType {
+			case schema.ArrayType:
+				if decData[key], err = jsoniter.MarshalToString(value); err != nil {
+					return nil, err
+				}
+			case schema.DateTimeType:
+				dateStr := fmt.Sprint(value)
```

Comment:
just do `dateStr, ok := value.(string)` otherwise, Sprint will do allocation which we don't need here.